tests/test_group.py

```diff
@@ -176,37 +176,86 @@ def test_set_item_intermediate(exdir_tmpfile):
     assert np.array_equal(exdir_tmpfile["group1/group2/group3/dataset"].data, np.array([1, 2, 3]))
 
 
-# TODO uncomment when deletion is implemented
 # Feature: Objects can be unlinked via "del" operator
-# def test_delete(setup_teardown_file):
-#     """Object deletion via "del"."""
-#
-#     f = setup_teardown_file[3]
-#     grp = f.create_group("test")
-#     grp.create_group("foo")
-#
-#     assert "foo" in grp
-#     del grp["foo"]
-#     assert "foo" not in grp
-#
-# def test_nonexisting(setup_teardown_file):
-#     """Deleting non-existent object raises KeyError."""
-#     f = setup_teardown_file[3]
-#     grp = f.create_group("test")
-#
-#     with pytest.raises(KeyError):
-#         del grp["foo"]
-#
-# def test_readonly_delete_exception(setup_teardown_file):
-#     """Deleting object in readonly file raises KeyError."""
-#     f = setup_teardown_file[3]
-#     f.close()
-#
-#     f = File(setup_teardown_folder[1], "r")
-#
-#     with pytest.raises(KeyError):
-#         del f["foo"]
+def test_delete_group(setup_teardown_file):
+    """Object deletion via "del"."""
+
+    f = setup_teardown_file[3]
+    grp = f.create_group("test")
+    grp.create_group("foo")
+
+    assert "foo" in grp
+    del grp["foo"]
+    assert "foo" not in grp
+
+
+def test_delete_group_from_file(setup_teardown_file):
+    """Object deletion via "del"."""
+
+    f = setup_teardown_file[3]
+    grp = f.create_group("test")
+
+    assert "test" in f
+    del f["test"]
+    assert "test" not in f
+
 
+def test_delete_raw(setup_teardown_file):
+    """Object deletion via "del"."""
+
+    f = setup_teardown_file[3]
+    grp = f.create_group("test")
+    grp.create_raw("foo")
+
+    assert "foo" in grp
+    del grp["foo"]
+    assert "foo" not in grp
+
+
+def test_nonexisting(setup_teardown_file):
+    """Deleting non-existent object raises KeyError."""
+    f = setup_teardown_file[3]
+    grp = f.create_group("test")
+
+    with pytest.raises(KeyError):
```

Comment:
Consider using `match` arg in `raises` to also check the error message

Reply:
good idea!